engine/src/main/java/org/terasology/engine/ComponentSystemManager.java

```diff
@@ -85,6 +85,14 @@ public void loadSystems(ModuleEnvironment environment, NetworkMode netMode) {
             RegisterSystem registerInfo = type.getAnnotation(RegisterSystem.class);
             if (registerInfo.value().isValidFor(netMode, isHeadless)) {
                 systemsByModule.put(moduleId, type);
+                if(registerInfo.requiresOptional().length != 0) {
```

Comment:
I think it would be good to put this in a new method - also to clarify what it does exactly. I find it hard to follow why it inserts it first and removes it immediately after if a condition is met.

Could this become a boolean method that decides whether to insert it or not?